src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -1,30 +1,154 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for license information.
 #---------------------------------------------------------------------------------------------
 
 import uuid
-import re
 
 from azure.cli._util import CLIError
+from azure.cli.commands.client_factory import get_mgmt_service_client, configure_common_settings
+from azure.mgmt.authorization import AuthorizationManagementClient
+from azure.graphrbac import GraphRbacManagementClient
 from azure.mgmt.authorization.models import RoleAssignmentProperties
-from ._params import _auth_client_factory
+from azure.graphrbac.models import UserCreateParameters, PasswordProfile
+
+def _auth_client_factory(**_):
+    return get_mgmt_service_client(AuthorizationManagementClient)
+
+def _graph_client_factory(**_):
+    from azure.cli._profile import Profile
+    profile = Profile()
+    cred, _, tenant_id = profile.get_login_credentials(True)
+    client = GraphRbacManagementClient(cred, tenant_id)
+    configure_common_settings(client)
+    return client
 
-#TODO: expand the support to be in parity with node cli
-def create_role_assignment(role, object_id, scope=None):
+def create_role_assignment(role, assignee, resource_group_name=None, resource_id=None):
+    '''
+    :param assignee: represent a user, group, or service principal.
+    supported format: object id, user sign-in name, or service principal name
+    :param resource_id: resource id
+    '''
     assignments_client = _auth_client_factory().role_assignments
     definitions_client = _auth_client_factory().role_definitions
-    role_id = role
-    scope = scope or '/subscriptions/' + definitions_client.config.subscription_id
-    if not re.match(r'[0-9a-f]{32}\Z', role, re.I): #retrieve role id
+
+    if resource_id:
+        if resource_group_name:
+            err = 'Resource group "{}" is redundant because resource id is supplied'
+            raise CLIError(err.format(resource_group_name))
+        scope = resource_id
+    else:
+        scope = '/subscriptions/' + definitions_client.config.subscription_id
+        if resource_group_name:
+            scope = scope + '/resourceGroups/' + resource_group_name
+
+    role_id = None
+    try:
+        uuid.UUID(role)
+        role_id = role
+    except ValueError:
+        pass
+
+    if not role_id: #retrieve role id
         role_defs = list(definitions_client.list(scope, "roleName eq '{}'".format(role)))
         if not role_defs:
-            raise CLIError('Role {} doesn\'t exist.'.format(role))
+            raise CLIError("Role '{}' doesn't exist.".format(role))
         elif len(role_defs) > 1:
-            raise CLIError('More than one roles match the given name {}'.format(role))
+            ids = [r.id for r in role_defs]
+            err = "More than one roles match the given name '{}'. Please pick a value from '{}'"
```

Comment:
>"More than one roles match the given name '{}'. Please pick a value from '{}'" [](start = 18, length = 78)

If I fall into this case, how would I resolve it with the error message that's provided?
Do I have to provide resource-id, set a different role, etc.?

If it would be clear for the context of the user of this command the action they need to take, it's fine. I was asking for clarification as I'm not sure the error message is the clearest.